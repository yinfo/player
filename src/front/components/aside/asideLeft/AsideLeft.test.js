// @flow

import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import AsideLeft from './AsideLeft';
import { navigation } from '../../../models/navigation';

// mock Horloge since snapshot would be different at each run
jest.mock('../../horloge/Horloge', () => 'Horloge');

const { sideMenu: sideMenus } = navigation;

describe('AsideLeft component', () => {
  it('renders as expected', () => {
    const component = shallow(
      <MemoryRouter>
        <AsideLeft
          isAnimated
          isCollapsed
          connectionStatus={{
            online: 'online',
            disconnected: 'disconnected',
          }}
          sideMenu={sideMenus}
          showPicture={false}
          userPicture={null}
          username="test"
          userIsConnected
          currentView={sideMenus[0].menus[0].name}
        />
      </MemoryRouter>,
    );
    expect(component).toMatchSnapshot();
  });
});
