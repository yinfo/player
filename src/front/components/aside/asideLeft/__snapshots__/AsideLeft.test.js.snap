// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsideLeft component renders as expected 1`] = `ShallowWrapper {}`;
